### DISCUSSION TITLE
How can create property which accept boolean?

### DISCUSSION THREAD

**Discussion by jitendraP-ashutec** [lit/lit#3625] [Q&A] [answered]
In lit there is property boolean, If user want to use it then he can add/remove attribute. but  I want know that lets say user can also able to pass property true/false as value to this property then How we can do that in lit ?

e.g 

```
var displayButton = true;

<wc-button  display="displayButton"> </wc-button>
```

so If use will toggle value from true to false button will be hide. If I use display type string then it will always show button event if I will pass 'true' or 'false' because it will consider it as string.

Any suggestion on it ?

Thanks

**Comment by p-ob** [comment, marked as answer]
Hey @jitendraP-ashutec!

Something that you might find useful is to stay consistent with the way the web treats boolean attributes and properties. I understand that frameworks might do something special, but staying consistent with the platform makes it less surprising when a consumer is jumping between different paradigms (like what if they just write vanilla HTML?).

So I'd recommend doing what vanilla HTML does:
```html
<wc-button id="button1"  display=""> </wc-button>
<script>
console.log(document.getElementById("button1").display); // logs "true"
</script>
```

```html
<wc-button id="button2"> </wc-button>
<script>
console.log(document.getElementById("button2").display); // logs "false"
</script>
```

```html
<wc-button id="button3" display="false"> </wc-button>
<script>
console.log(document.getElementById("button3").display); // logs "true"
</script>
```

Essentially, the property is true as long as element.hasAttribute("display") is true. It doesn't matter what value is between the quotation marks. I get you want to provide that ergonomics, but I've found that sticking with what the platform does is the most manageable long term.

All that being said.... if you _really_ want to do this, you could provide [a custom converter](https://lit.dev/docs/components/properties/#conversion-converter) to your property declaration.

**Comment by augustjk** [comment]
If you're talking about defining a boolean property/attribute on a component, you can do that by setting the property type.

```js
class WcButton extends LitElement {
  static properties = {
    display: {type: Boolean}
  }

  constructor() {
    super();
    this.display = false;
  }
}
```

or

```ts
class WcButton extends LitElement {
  @property({type: Boolean})
  display = false;
}
```

You need to make sure it defaults to false. See https://lit.dev/docs/components/properties/#boolean-attributes

To then provide a boolean attribute within a `lit-html` template, you can use the `?` prefix.

```js
html`<wc-button ?display=${displayButton}></wc-button>`;
```

See https://lit.dev/docs/templates/expressions/#boolean-attribute-expressions
